internal/app/singularity/push.go

```diff
@@ -97,5 +103,17 @@ func LibraryPush(file, dest, authToken, libraryURI, keyServerURL, remoteWarning
 	}
 	defer f.Close()
 
-	return libraryClient.UploadImage(context.Background(), f, r.Host+r.Path, r.Tags, "No Description", &progressCallback{})
+	return libraryClient.UploadImage(context.Background(), f, r.Host+r.Path, arch, r.Tags, "No Description", &progressCallback{})
```

Comment:
funny, I would have expected the library to look inside the file to figure out the architecture, not that the client has to tell it that information.

Reply:
It does - see just above where `arch` comes from...

```
arch, err := sifArch(file)
```

Reply:
<edit> ahh I see what you mean...</edit>

The library does look into the file after upload. Since things go into S3 directly now this has to be async, after the upload completed. In order the arch->tag->Image is correct immediately we have to send it here.